src/sortStudents.ts

```diff
@@ -1,16 +1,78 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'Name',
+  Surname = 'Surname',
+  Age = 'Age',
+  Married = 'Married',
+  AverageGrade = 'AverageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+function calculateAverageGrade(grades: number[]): number {
+  const sum = grades.reduce((acc, grade) => acc + grade, 0);
+
+  return sum / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents = [...students];
+
+  sortedStudents.sort((a, b) => {
+    let aValue;
+    let bValue;
+
+    switch (sortBy) {
+      case SortType.Name:
+        aValue = a.name;
+        bValue = b.name;
+        break;
+
+      case SortType.Surname:
+        aValue = a.surname;
+        bValue = b.surname;
+        break;
```

Comment:
```suggestion
      case SortType.Name:
      case SortType.Surname:
        aValue = a[sortBy];
        bValue = b[sortBy];
        break;
```
we can do the same with age and avg grade